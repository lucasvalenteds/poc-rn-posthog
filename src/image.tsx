import React from 'react';
import Native from 'react-native';

export type ImageType = {
  testID?: string;
  uri?: string;
};

export const Image: React.FC<ImageType> = (props) => {
  const window = Native.useWindowDimensions();

  return (
    <React.Fragment>
      <Native.View
        testID={props.testID}
        style={{
          elevation: 4,
          shadowColor: '#000000',
          shadowOffset: { width: 0, height: 4 },
          shadowOpacity: 0.5,
          shadowRadius: 4,
        }}>
        <Native.Image
          style={{
            width: window.width / 1.2,
            height: window.width / 1.2,
            borderRadius: 4,
          }}
          source={{
            uri: props.uri,
            width: window.width / 1.2,
            height: window.width / 1.2,
          }}
        />
      </Native.View>
    </React.Fragment>
  );
};
