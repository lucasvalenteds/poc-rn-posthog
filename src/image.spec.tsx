import React from 'react';
import * as Testing from 'react-native-testing-library';

import { Image } from './image';

describe('Image', () => {
  test('Matching snapshot', () => {
    const component = Testing.render(<Image uri={''} />);

    expect(component).toMatchSnapshot();
  });
});
