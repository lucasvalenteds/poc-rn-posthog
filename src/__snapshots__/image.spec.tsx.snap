// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Matching snapshot 1`] = `
<View
  style={
    Object {
      "elevation": 4,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 4,
    }
  }
>
  <Image
    source={
      Object {
        "height": 625,
        "uri": "",
        "width": 625,
      }
    }
    style={
      Object {
        "borderRadius": 4,
        "height": 625,
        "width": 625,
      }
    }
  />
</View>
`;
